cpp/src/arrow/compute/kernels/select_k_test.cc

```diff
@@ -42,7 +42,7 @@ using internal::checked_pointer_cast;
 namespace compute {
 
 template <typename ArrayType, SortOrder order>
-class SelectKComparator {
+class SelectKCompareForResult {
  public:
```

Comment:
I still do not see what is the difference between `SelectKComparatorForResult` and `SelectKComparator`.

Reply:
SelectKComparatorForResult consider duplicates to check duplicates in the result that's why it uses <=, the other it only uses `<` as a comparator in the Heap.